src/lib.rs

```diff
@@ -0,0 +1,23 @@
+use thiserror::Error;
+
+mod big_int;
+mod macros;
+
+pub use crate::big_int::BigInt;
+pub use crate::big_int::RNSRepresentation;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum CrateError {
```

Comment:
This name is too generic IMO. Let's use something more specific to this crate.

Reply:
Is `BigIntError` okay for you ?